tests/phpunit/tests/media.php

```diff
@@ -3462,6 +3457,161 @@ public function data_wp_get_attachment_image_decoding_attr() {
 		);
 	}
 
+	/**
+	 * Tests that the `wp_get_loading_optimization_attributes()` function should add
+	 * the 'decoding' attribute.
+	 *
+	 * @ticket 58892
+	 *
+	 * @dataProvider data_add_decoding_attr
+	 *
+	 * @param string $tag_name The tag name.
+	 * @param string $decoding The value for the 'decoding' attribute. 'no value' for default.
+	 * @param string $expected The expected `img` tag.
+	 * @param string $message  The test message.
+	 */
+	public function test_add_decoding_attr( $tag_name, $decoding, $expected, $message ) {
+		add_filter(
+			'wp_img_tag_add_decoding_attr',
+			static function ( $value ) use ( $decoding ) {
+				return $decoding;
+			}
+		);
+
+		$this->assertEquals(
+			$expected,
+			wp_get_loading_optimization_attributes(
+				$tag_name,
+				array(
+					'width'  => 1,
+					'height' => 1,
+				),
+				''
+			),
+			$message
+		);
+	}
+
+	/**
+	 * Data provider for test_add_decoding_attr().
+	 *
+	 * @return array {
+	 *  @type array $test {
+	 *      @type string $tag_name The tag name.
+	 *      @type string $decoding The value to test. Additional context to pass to the filters.
+	 *      @type array  $expected The expected value.
+	 *  }
+	 * }
+	 */
+	public function data_add_decoding_attr() {
+		return array(
+			// Happy path.
+			'async'                   => array(
+				'tag_name' => 'img',
+				'decoding' => 'async',
+				'expected' => array(
+					'decoding' => 'async',
+					'loading'  => 'lazy',
+				),
+				'message'  => 'The decoding attribute should accept `async` string.',
+			),
+			'sync'                    => array(
+				'tag_name' => 'img',
+				'decoding' => 'sync',
+				'expected' => array(
+					'decoding' => 'sync',
+					'loading'  => 'lazy',
+				),
+				'message'  => 'The decoding attribute should accept `sync`string.',
+			),
+			'auto'                    => array(
+				'tag_name' => 'img',
+				'decoding' => 'auto',
+				'expected' => array(
+					'decoding' => 'auto',
+					'loading'  => 'lazy',
+				),
+				'message'  => 'The decoding attribute should accept `auto` string.',
+			),
+
+			// Unhappy paths.
+			'lazy (unaccepted value)' => array(
+				'tag_name' => 'img',
+				'decoding' => 'lazy',
+				'expected' => array(
+					'loading' => 'lazy',
+				),
+				'message'  => 'The decoding attribute should not accept `lazy` string.',
+			),
+			'a non-string value'      => array(
+				'tag_name' => 'img',
+				'decoding' => array( 'sync' ),
+				'expected' => array(
+					'loading' => 'lazy',
+				),
+				'message'  => 'The decoding attribute should not accept non-string values.',
+			),
+
+			// Falsy values.
```

Comment:
See my feedback in https://github.com/WordPress/wordpress-develop/pull/4991#discussion_r1309472208: Falsey values are and have historically been a way to indicate "omit the attribute", which is something that we have to maintain. Basically, if someone passes `$attr['decoding'] = false`, we should leave that untouched and not overwrite the `decoding` attribute to become `async` like we usually do. And then, before actually applying the attributes to the tag, we need to have a check where we `unset( $attr['decoding'] )` if it is false-y.